<?xml version="1.0"?>
<!-- Also Note that these scripts are meant to be used with a 
launch driver. In this case im using the interbotix nav stack -->
<!-- command is: roslaunch interbotix_xslocobot_nav xslocobot_nav_2.launch 
robot_model:=locobot_wx250s use_lidar:=true localization:=true-->
<launch>  
    <!-- Global Arguments -->
    <arg name="robot_name" default="locobot"/>
    <arg name="rviz_frame" default="map"/>
    <arg name="rvizconfig" default="$(find yolo_vision)/rviz/xslocobot_description_yolo.rviz"/>
    <arg name="poses_config" default="$(find move_to_pose)/config/poses.yaml"/>
    <arg name="pose_command" default="all"/>
    <arg name="use_rviz" default="true" doc="Set to false to disable RViz visualization"/>
    
    <!-- Move Base Topics -->
    <arg name="move_base_topic" default="/$(arg robot_name)/move_base"/>
    <arg name="costmap_topic" default="/$(arg robot_name)/move_base/global_costmap/costmap"/>
    
    <!-- Object Detection Topics -->
    <arg name="object_marker_topic" default="/locobot/object_markers"/>
    <arg name="bbox_depth_topic" default="camera/yolo/bbox_depth"/>
    
    <!-- Include the object mapper launch file -->
    <include file="$(find yolo_vision)/launch/object_mapper.launch">
        <arg name="robot_name" value="$(arg robot_name)"/>
        <arg name="rviz_frame" value="$(arg rviz_frame)"/>
        <arg name="rvizconfig" value="$(arg rvizconfig)"/>
        <arg name="object_marker_topic" value="$(arg object_marker_topic)"/>
        <arg name="bbox_depth_topic" value="$(arg bbox_depth_topic)"/>
        <arg name="use_rviz" value="false"/>
    </include>
    
    <!-- Launch Scan Detect Approach Node -->
    <node name="scan_and_approach_node" pkg="move_to_pose" type="scan_and_approach_node.py" output="screen">
        <!-- Common parameters -->
        <param name="robot_name" value="$(arg robot_name)"/>
        <param name="poses_config" value="$(arg poses_config)"/>
        <param name="pose_command" value="$(arg pose_command)"/>
        <param name="move_base_topic" value="$(arg move_base_topic)"/>
        <param name="costmap_topic" value="$(arg costmap_topic)"/>
    
        
        <!-- Approach object parameters -->
        <param name="object_marker_topic" value="$(arg object_marker_topic)"/>
        <param name="bbox_depth_topic" value="/$(arg robot_name)/$(arg bbox_depth_topic)"/>
    </node>
    
    <!-- RViz (conditionally launched) -->
    <group if="$(arg use_rviz)">
        <node name="rviz" pkg="rviz" type="rviz" 
              args="-f $(arg rviz_frame) -d $(arg rvizconfig)"
              ns="$(arg robot_name)">
            <remap from="/clicked_point" to="clicked_point"/>
            <remap from="/initialpose" to="initialpose"/>
            <remap from="/move_base_simple/goal" to="move_base_simple/goal"/>
        </node>
    </group>
</launch>